From d42a89d626b75511b057061cfc41e9dde8db623d Mon Sep 17 00:00:00 2001
From: Simon Glass <sjg@chromium.org>
Date: Fri, 10 Feb 2023 12:12:07 -0700
Subject: [PATCH] Avoid build failures with dumpRSAPublicKey on recent distros

Emit a warning and continue when OpenSSL is newer than expected. The
chroot still uses an older version.

This was broken by:

   0ca7a9e4 ("firmware: host: futility: Add CBFS metadata hash support")

BUG=b:245993083, b:246328810, b:197114807
BRANCH=none
TEST=make -C ~/cosarm/src/platform/vboot_reference/ USE_FLASHROM=0
See that the errors become warnings and the build completes

Change-Id: Id4e25e4ebad1d9ae1f6ee5425b3e6fe3cbd5d5e5
Signed-off-by: Simon Glass <sjg@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/platform/vboot_reference/+/4241424
Reviewed-by: Yu-Ping Wu <yupingso@chromium.org>
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index a4390522..8cc68e3c 100644
--- a/Makefile
+++ b/Makefile
@@ -1016,6 +1016,11 @@ util_files: $(if ${SDK_BUILD},${UTIL_FILES_SDK},${UTIL_FILES_BOARD})
 # These have their own headers too.
 ${BUILD}/utility/%: INCLUDES += -Iutility/include
 
+# Avoid build failures outside the chroot on Ubuntu 2022.04
+ifeq ($(OPENSSL_VERSION),3)
+${BUILD}/utility/%: CFLAGS += -Wno-error=deprecated-declarations
+endif
+
 ${UTIL_BINS_SDK}: ${UTILLIB}
 ${UTIL_BINS_SDK}: LIBS = ${UTILLIB}
 ${UTIL_BINS_BOARD}: ${UTILLIB}
-- 
2.39.1

